src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Navigate } from "react-router-dom";
+import { useAuth } from "../hooks/useAuth";
+
+// Define the props for ProtectedRoute
+interface ProtectedRouteProps {
+  children: React.ReactNode;
+}
+
+const ProtectedRoute: React.FC<ProtectedRouteProps> = ({ children }) => {
+  const { isAuthenticated } = useAuth();
+
+  if (isAuthenticated === null) {
+    return <div>Loading...</div>;
```

Comment:
Forever loading is not good, so we probably want to enforce redirect on whatever component gets wrapped. I'm sure this is handled elsewhere in the app already, but might be nice to keep it all in one place

Reply:
isAuthenticated is a hook that will return true or false once the authentication is checked. null is just what it is set to before that is determined.

Reply:
in failure cases (like no network), will it eventually error out and show something besides waiting?